Update the tests for Proxy::Util::CommandTask

sodabrew opened the pull request:
The `wget` test was hanging for me - I'm running OS X, and `127.0.0.2` is not a default alias of localhost. I couldn't figure out why the test was using `wget` and neither could the FIXME author. So in this update, use `true` and `false` to test $? return codes 0 and 1, respectively.

That said - Ruby 1.8.7 is heading out the door. If Proxy::Util::CommandTask cannot capture $? on current versions of Ruby, that's problem!

sodabrew commented:
Second commit should return subprocess exit status in all Rubies. I can squash the commits if this looks good.

domcleal commented:
I've added you to our GitHub org, so if you set membership to public (https://github.com/orgs/theforeman/members) then Jenkins will test your PRs automatically.  Thanks for the contributions!

[test]

sodabrew commented:
Sadly, Ruby 1.8 always returns 0 for popen3 - that's what the bug report link was about it just took me three times reading it to grok - but at least I've fixed status handling for newer Rubies!

Cool thanks! Let's try this!

[test]

lzap commented on lib/proxy/util.rb:
```diff
@@ -20,17 +20,21 @@ def initialize(cmd)
+          status = thr.value unless thr.nil?
         end
-        $?
+        status ? status.exitstatus : $?
```
Beware, AFAIK existstatus is not set until you close the file handle.

lzap replied:
Oh I've misread the indentation, you are safe :-)

lzap commented:
Sorry you will need to rebase :-( I've just merged Dmitri's patch which refactored this completely. http://git.io/zoPmbQ

sodabrew commented:
Rebased!

sodabrew committed:
Update Proxy::Util::CommandTask to handle Ruby 1.8 and 1.9+

In this update, use 'true' and 'false' to test $? return codes 0 and 1, respectively.
In Ruby 1.8.x, the return code from Open3.popen3 is always 0, some errors will not be caught.

domcleal commented:
Merged as bbafbc0 for 1.6.0, thanks again @sodabrew!
